usaspending_api/etl/management/commands/elasticsearch_indexer.py

```diff
@@ -3,232 +3,211 @@
+            logger.error(f"Write alias '{config['write_alias']}' is missing")
             raise SystemExit(1)
         # Force manual refresh for atomic transaction-like delete/re-add consistency during incremental load.
         # Turned back on at end.
         toggle_refresh_off(es_client, config["index_name"])
     else:
         if es_client.indices.exists(config["index_name"]):
-            logger.error(format_log(f"Data load into existing index. Change index name or run an incremental load"))
+            logger.error(f"Data load into existing index. Change index name or run an incremental load")
             raise SystemExit(1)
 
-    if not config["directory"].is_dir():
-        logger.error(format_log(f"Provided directory does not exist"))
+    if config["starting_date"] < default_datetime:
+        logger.error(f"--start-datetime is too early. Set no earlier than {default_datetime}")
         raise SystemExit(1)
-    elif config["starting_date"] < default_datetime:
-        logger.error(format_log(f"--start-datetime is too early. Set no earlier than {default_datetime}"))
-        raise SystemExit(1)
-    elif not config["is_incremental_load"] and config["process_deletes"]:
-        logger.error(format_log("Skipping deletions for ths load, --deleted overwritten to False"))
-        config["process_deletes"] = False
-
-    config["ingest_wait"] = options["idle_wait_time"]
 
     return config
 
 
-def set_config(copy_args: list, arg_parse_options: dict) -> dict:
+def set_config(passthrough_values: list, arg_parse_options: dict) -> dict:
     """Set values based on env vars and when the script started"""
-    root_index = settings.ES_TRANSACTIONS_QUERY_ALIAS_PREFIX
-    if arg_parse_options["load_type"] == "awards":
-        root_index = settings.ES_AWARDS_QUERY_ALIAS_PREFIX
-    config = {
-        "aws_region": settings.USASPENDING_AWS_REGION,
-        "s3_bucket": settings.DELETED_TRANSACTION_JOURNAL_FILES,
-        "root_index": root_index,
-        "processing_start_datetime": datetime.now(timezone.utc),
-        "verbose": arg_parse_options["verbosity"] > 1,  # convert the management command's levels of verbosity to a bool
-    }
-
-    config.update({k: v for k, v in arg_parse_options.items() if k in copy_args})
-    return config
 
+    if arg_parse_options["load_type"] == "awards":
+        config = {
+            "data_transform_func": transform_award_data,
+            "data_type": "award",
+            "max_query_size": settings.ES_AWARDS_MAX_RESULT_WINDOW,
+            "primary_key": "award_id",
+            "query_alias_prefix": settings.ES_AWARDS_QUERY_ALIAS_PREFIX,
+            "required_index_name": settings.ES_AWARDS_NAME_SUFFIX,
+            "sql_view": settings.ES_AWARDS_ETL_VIEW_NAME,
+            "stored_date_key": "es_awards",
+            "unique_id_field": "generated_unique_award_id",
+            "write_alias": settings.ES_AWARDS_WRITE_ALIAS,
+        }
+    else:
```

Comment:
Similar to pervious comment -- with expecting new load_types (like file c accounts data), best to be explicit here:

```
if awards:
    ...
elif transactions:
    ...
else: 
    raise ...
```

Reply:
ok